impl/src/main/java/slack/SlackResourceImpl.java

```diff
@@ -30,52 +37,111 @@ public SlackResourceImpl(SlackConfig slackConfig) {
         this.slackConfig = slackConfig;
     }
 
+    static String handleUserInfoResponse(UsersInfoResponse usersInfoResponse) {
+        if (usersInfoResponse.isOk()) {
+            if (usersInfoResponse.getUser().getProfile().getEmail() == null) {
+                LOG.warn("Could not fetch users email, maybe scope users:read.email is missing");
+                throw new IllegalStateException("User email from slack is null and might be due to missing scope users:read.email");
+            }
+            return usersInfoResponse.getUser().getProfile().getEmail();
+        } else {
+            throw new IllegalStateException("Could not fetch user email from slack: " + usersInfoResponse.getError());
+        }
+    }
+
+    static String handleUserByEmailResponse(UsersLookupByEmailResponse usersLookupByEmailResponse) {
+        if (!usersLookupByEmailResponse.isOk()) {
+            throw new IllegalStateException("Could not get user by email from slack: " + usersLookupByEmailResponse.getError());
+        }
+
+        return usersLookupByEmailResponse.getUser().getId();
+    }
+
+    static Observable<Void> handleChatPostMessageResponse(ChatPostMessageResponse chatPostMessageResponse) {
+        if (!chatPostMessageResponse.isOk()) {
+            LOG.warn("Could not post to slack: {}", chatPostMessageResponse.getError());
+            return error(new IllegalStateException(chatPostMessageResponse.getError()));
+        }
+        return empty();
+    }
+
+    static Message handleGetMessageResponse(ChannelsRepliesResponse channelsRepliesResponse) {
+        if (channelsRepliesResponse.isOk()) {
+            return channelsRepliesResponse.getMessages().get(0);
+        }
+        LOG.warn("Could not get message from slack: {}", channelsRepliesResponse.getError());
+        throw new IllegalStateException(channelsRepliesResponse.getError() + " " + channelsRepliesResponse.getWarning());
+    }
+
     @Override
     public Observable<String> getUserEmail(String userId) {
-        return fromCallable(() -> slack.methods().usersInfo(
-            UsersInfoRequest
-            .builder()
+        return callSlack(() -> {
+
+            final UsersInfoRequest slackRequest = UsersInfoRequest
+                .builder()
                 .user(userId)
                 .token(slackConfig.getApiToken())
-            .build())
-            .getUser()
-            .getProfile()
-            .getEmail());
+                .build();
+
+            return slack.methods().usersInfo(slackRequest);
+        }).map(SlackResourceImpl::handleUserInfoResponse);
+    }
+
+    private <T> Observable<T> callSlack(Callable<T> callable) {
+        return fromCallable(callable);
```

Comment:
jSlack uses OkHttp which should have [pooled connections](https://square.github.io/okhttp/connections/) but we still block on the rxnetty eventloop waiting for an result. Perhaps use Schedulers.io() (unbounded) or a dedicated threadpool?